src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -0,0 +1,462 @@
+import { databaseNodeIcon } from '../../../../components/workflows/nodes/DatabaseNode';
+import { dictArtifactNodeIcon } from '../../../../components/workflows/nodes/DictArtifactNode';
+import { functionOperatorNodeIcon } from '../../../../components/workflows/nodes/FunctionOperatorNode';
+import { genericArtifactNodeIcon } from '../../../../components/workflows/nodes/GenericArtifactNode';
+import { imageArtifactNodeIcon } from '../../../../components/workflows/nodes/ImageArtifactNode';
+import { jsonArtifactNodeIcon } from '../../../../components/workflows/nodes/JsonArtifactNode';
+import { metricOperatorNodeIcon } from '../../../../components/workflows/nodes/MetricOperatorNode';
+import { numericArtifactNodeIcon } from '../../../../components/workflows/nodes/NumericArtifactNode';
+import { stringArtifactNodeIcon } from '../../../../components/workflows/nodes/StringArtifactNode';
+import { tableArtifactNodeIcon } from '../../../../components/workflows/nodes/TableArtifactNode';
+import { NodeType } from '../../../../reducers/nodeSelection';
+import {
+  handleGetArtifactResults,
+  handleGetOperatorResults,
+  handleGetWorkflow,
+} from '../../../../reducers/workflow';
+import { AppDispatch, RootState } from '../../../../stores/store';
+import { ArtifactType } from '../../../../utils/artifacts';
+import UserProfile from '../../../../utils/auth';
+import { Data } from '../../../../utils/data';
+import { getPathPrefix } from '../../../../utils/getPathPrefix';
+import { LoadingStatusEnum } from '../../../../utils/shared';
+import DefaultLayout from '../../../layouts/default';
+import { LayoutProps } from '../../types';
+
+type MetricDetailsHeaderProps = {
+  artifactName: string;
+  createdAt?: string;
+};
+
+const MetricDetailsHeader: React.FC<MetricDetailsHeaderProps> = ({
+  artifactName,
+  createdAt,
+}) => {
+  return (
+    <Box width="100%" display="flex" alignItems="center">
+      <FontAwesomeIcon
+        height="24px"
+        width="24px"
+        style={{ marginRight: '8px' }}
+        icon={faCircleCheck}
+        color={'green'}
+      />
+      <Typography variant="h4" component="div">
+        {artifactName}
+      </Typography>
+      {createdAt && (
+        <Typography marginTop="4px" variant="caption" component="div">
+          Created: {createdAt}
+        </Typography>
+      )}
+    </Box>
+  );
+};
+
+type MetricDetailsPageProps = {
+  user: UserProfile;
+  Layout?: React.FC<LayoutProps>;
+};
+
+const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
+  user,
+  Layout = DefaultLayout,
+}) => {
+  const dispatch: AppDispatch = useDispatch();
+  const navigate = useNavigate();
+  const { workflowId, workflowDagResultId, metricOperatorId } = useParams();
+
+  const [inputsExpanded, setInputsExpanded] = useState<boolean>(true);
+  const [outputsExpanded, setOutputsExpanded] = useState<boolean>(true);
+
+  const workflow = useSelector((state: RootState) => state.workflowReducer);
```

Comment:
@likawind after we get this PR merged in, we can start incorporating your new api route for getting results populated in the dag structure.